packages/mobile/src/components/core/SegmentedControl.tsx

```diff
@@ -0,0 +1,211 @@
+import { Fragment, useState, useEffect, useRef } from 'react'
+
+import {
+  Animated,
+  LayoutChangeEvent,
+  Pressable,
+  TextStyle,
+  View,
+  ViewStyle,
+  Text
+} from 'react-native'
+
+import { makeStyles, StylesProps } from 'app/styles'
+
+// Note, offset is the inner padding of the container div
+const OFFSET = 3
+
+export type Option = {
+  key: string
+  text: string
+}
+
+export type SegmentedControlProps = {
+  // The options to display for the tab slider
+  options: Array<Option>
+
+  // Key of selected option
+  selected?: string
+
+  // Key of initially selected option
+  defaultSelected?: string
+
+  // Callback fired when new option is selected
+  onSelectOption: (key: string) => void
+
+  fullWidth?: boolean
+} & StylesProps<{
+  root: ViewStyle
+  tab: ViewStyle
+  text: TextStyle
+  activeText: TextStyle
+}>
+
+const springToValue = (
+  animation: Animated.Value,
+  value: number,
+  finished?: () => void
+) => {
+  Animated.spring(animation, {
+    toValue: value,
+    tension: 160,
+    friction: 15,
+    useNativeDriver: false
+  }).start(finished)
+}
+
+const useStyles = makeStyles(({ palette, typography, spacing }) => ({
+  tabs: {
+    borderRadius: 6,
+    backgroundColor: palette.neutralLight7,
+    flexDirection: 'row',
+    alignItems: 'center',
+    padding: OFFSET,
+    paddingRight: 0
+  },
+  tab: {
+    elevation: 3,
+    paddingTop: 10,
+    paddingHorizontal: spacing(4),
```

Comment:
why use spacing here and not for `paddingTop`?

Reply:
Missed it good catch